ONSdigital/ssdc-rm-support-tool
# Originating user for new case events are the person who clicked process

# Motivation and Context

Support Tool user who clicked "Process" sample button should be originating user of newCase events
# What has changed



New case events have the user who pressed process as the originating user.
Job details pop up will say who either processed it or cancelled the job.
# How to test?



To check the processed by stuff:

1. Build the image and `make up` in docker dev
2. Stop the case processor in docker dev (`docker stop caseprocessor`)
3. Load a sample, let it validate but don't process immediately.
4. Change the user you're working as.
5. Process the sample file
6. Check the user you're acting as and the time you clicked process show up on the job details window
7. In the docker dev repo ` ./get_message.sh event_new-case_rm-case-processor shared-project` and check the originating user in the JSON is the user you changed to to process the job.

To check the cancelled by stuff:

1. Do as above up to step 5 (no need to stop the case processor)
2. Go into the jobs detail page and click cancel
3. Check that the user you're acting as and the time you clicked cancel show up on the job details window

# Links



https://trello.com/c/DMFLX8em/2878-support-tool-user-who-clicked-process-sample-button-should-be-originating-user-of-newcase-events-8
# Screenshots (if appropriate):
